src/pay/batch_payer.py

```diff
@@ -687,9 +687,22 @@ def attempt_single_batch(self, payment_items, op_counter, dry_run=None):
 
             # TRD extension for non scriptless contract accounts
             if payment_item.paymentaddress.startswith("KT"):
-                simulation_status, simulation_results = self.simulate_single_operation(
-                    payment_item, pymnt_amnt, branch, chain_id
-                )
+                try:
+                    (
+                        simulation_status,
+                        simulation_results,
+                    ) = self.simulate_single_operation(
+                        payment_item, pymnt_amnt, branch, chain_id
+                    )
+                except Exception as e:
+                    logger.info(
+                        "Payment to {} script could not be processed. Payment simulation failed with error: {}: {} ".format(
+                            payment_item.paymentaddress, type(e).__name__, str(e)
+                        )
+                    )
+                    payment_item.paid = PaymentStatus.FAIL
+                    payment_item.desc += "Payment simulation encountered an error while executing. Marking payment as failed. "
+                    continue
 
                 if simulation_status == PaymentStatus.FAIL:
```

Comment:
`simulation_status` can be `FAILED` if `status != HTTPStatus.OK`. I agree with you that if the node/api is not reachable, that should not `AVOID` the payment, but mark it as failed so that it can be retried again when the node/api is available again.

Reply:
If that is a fair assumption, then we could throw an exception when `status != HTTPStatus.OK` and it will be marked as failed instead of avoided.

Reply:
now there is still the mystery of double entires. As this catch was missing the exception was obviously cought one level above and it retried the whole batch. Maybe it is due to this?

Reply:
@AndrewKishino should we progress here? Do you want to finish it?